src/main/java/org/broadinstitute/hellbender/tools/spark/CreateIndexForHadoopBam.java

```diff
@@ -0,0 +1,158 @@
+        try(BufferedInputStream in = new BufferedInputStream(new FileInputStream(inputBam))){
+            try(BufferedOutputStream out = new BufferedOutputStream(new FileOutputStream(index))) {
+                SplittingBAMIndexer.index(in, out, inputBam.length(), granularity);
+            } catch (IOException e){
+                throw new UserException.CouldNotCreateOutputFile(index, e);
+            }
+        } catch (IOException e) {
+            throw new UserException.CouldNotReadInputFile(inputBam, e);
+        }
+    }
+
+    private static void createBaiAndSplittingIndex(File inputBam, File index, int granularity, ValidationStringency readValidationStringency) {
+        assertIsBam(inputBam);
+        try(SamReader reader = SamReaderFactory.makeDefault()
+                .validationStringency(readValidationStringency)
+                .setOption(SamReaderFactory.Option.INCLUDE_SOURCE_IN_RECORDS, true)
+                .open(inputBam)){
+            try(BufferedOutputStream out = new BufferedOutputStream(new FileOutputStream(index))){
+                final SAMFileHeader header = reader.getFileHeader();
+                assertBamIsCoordinateSorted(header);
+                SplittingBAMIndexer indexer = new SplittingBAMIndexer(out, granularity);
+
+                BAMIndexer bamIndexer = new BAMIndexer(IOUtils.replaceExtension(index, BAMIndex.BAMIndexSuffix), header);
+                for(SAMRecord read : reader){
+                    indexer.processAlignment(read);
+                    bamIndexer.processAlignment(read);
+                }
+                indexer.finish(inputBam.length());
+                bamIndexer.finish();
+            } catch (final IOException e){
+                throw new UserException.CouldNotCreateOutputFile(index, e);
+            }
+        } catch (final IOException e) {
+            throw new UserException.CouldNotReadInputFile(inputBam, e);
+        }
+    }
+
+    private static void assertBamIsCoordinateSorted(SAMFileHeader header) {
+        if( header.getSortOrder() != SAMFileHeader.SortOrder.coordinate) {
+            throw new UserException.BadInput("Cannot create a " + BAMIndex.BAMIndexSuffix + " index for a file " +
+                    "that isn't coordinate sorted.");
+        }
+    }
+
+
+    private static void assertIsBam(File inputBam) {
+        if(!BamFileIoUtils.isBamFile(inputBam)) {
+            throw new UserException.BadInput("A splitting index is only relevant for a bam file, but a "
+                    + "file with extension "+ FileUtils.getExtension(inputBam.getName()) + " was specified.");
+        }
+    }
+
+    private static File getOutputFile(File output, File input) {
+        if(output == null){
+            return new File(input.getPath() + SplittingBAMIndexer.OUTPUT_FILE_EXTENSION);
+        } else {
+            if (!output.toString().endsWith("bam" + SplittingBAMIndexer.OUTPUT_FILE_EXTENSION)){
```

Comment:
dont use toString() for any logic like this. Do you mean `input.getAbsolutePath()` or something?